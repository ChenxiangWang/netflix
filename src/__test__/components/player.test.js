import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import Player from "../../components/player";

describe("<Player>", () => {
  // it = test(name, fn, timeout);
  it("render the <Player> with a bunny video", () => {
    const { container } = render(
      <Player>
        <Player.Button />
        <Player.Video src={"/videos/bunny.mp4"} data-testid="player" />
      </Player>
    );
    expect(screen.queryByTestId("player")).toBeFalsy();
    fireEvent.click(screen.getByText("Play"));

    expect(screen.getByTestId("player")).toBeTruthy();
    fireEvent.click(screen.queryByTestId("player"));

    expect(screen.queryByTestId("player")).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
