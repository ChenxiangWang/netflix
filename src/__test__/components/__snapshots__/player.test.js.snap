// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player> render the <Player> with a bunny video 1`] = `
<div
  class="style__Container-sc-1wkdspn-0"
>
  <button
    class="style__Button-sc-1wkdspn-4 fYiLbt"
  >
    Play
  </button>
</div>
`;
